applications/asset_tracker_v2/doc/modem_module.rst

```diff
@@ -0,0 +1,149 @@
+If a FOTA download fails, a ``MODEM_EVT_CARRIER_FOTA_STOPPED`` event is sent, and the application may again establish TLS connections and continue normal operation.
+
+For more details and configuration options, you can refer to :ref:`liblwm2m_carrier_readme`.
+
+Modem information
+=================
+
+When the application module sends out an ``APP_EVT_DATA_GET``, the modem module will check the requested data list for relevant requests:
+
+* ``APP_DATA_MODEM_STATIC`` - Static modem data, such as configured system mode (any combination of LTE-M, NB-IoT and GNSS), ICCID, modem firmware version, application version and board version.
+* ``APP_DATA_MODEM_DYNAMIC`` - Dynamic modem data, such as Cell ID, tracking area code, RSRP, IP address and PLMN (MCCMNC).
+* ``APP_DATA_BATTERY`` - Voltage of the the modem's power supply.
+
+The module uses :ref:`modem_info_readme` to  acquire information about the modem, LTE network and the modem's power supply.
+The response for the three different data types are sent as separate events:
+
+* Static data as ``MODEM_EVT_MODEM_STATIC_DATA_READY``:
+* Dynamic data as ``MODEM_EVT_MODEM_DYNAMIC_DATA_READY``:
+* Battery voltage as ``MODEM_EVT_BATTERY_DATA_READY``
+
+If sampling of data fails, a corresponding error message is sent: ``MODEM_EVT_MODEM_STATIC_DATA_NOT_READY``, ``MODEM_EVT_MODEM_DYNAMIC_DATA_NOT_READY`` or ``MODEM_EVT_BATTERY_DATA_NOT_READY``.
+
+Neighbor cell measurements
+==========================
+
+Neighbor measurements can be requested by the application by sending an ``APP_EVT_DATA_GET`` event where ``APP_DATA_NEIGHBOR_CELLS`` is part of the requested data list.
+Upon reception of that event, the modem module uses the LTE link controller to start a neighbor search of type ``LTE_LC_NEIGHBOR_SEARCH_TYPE_DEFAULT``. You can refer to :ref:`lte_lc_readme` for more details on the available search types.
+When the search completes, the module sends a ``MODEM_EVT_NEIGHBOR_CELLS_DATA_READY`` event containing the cell information received from the modem.
+If the search fails, a ``MODEM_EVT_NEIGHBOR_CELLS_DATA_NOT_READY`` event is sent.
+
+Module internals
+****************
+
+The modem module has an internal thread with a message queue for processing.
+When an event is received in the :ref:`event_manager` handler, the event is converted to a message and put into the module's queue for processing in thread context.
+This gives the module the flexibility to call functions that may take some time to complete.
+
+Module states
+=============
+
+The modem module has an internal state machine with the following states:
+
+  * ``STATE_INIT`` - The initial state of the module in which it awaits the modem to be initialized.
+    This module will enter this state only if the :ref:`liblwm2m_carrier_readme` library is enabled, as the modem is automatically initialized otherwise.
```

Comment:
```suggestion
    This module will enter this state only if the :ref:`liblwm2m_carrier_readme` library is enabled, because the modem is automatically initialized otherwise.
```

Reply:
fixed